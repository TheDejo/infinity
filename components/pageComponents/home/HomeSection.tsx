import React from 'react';
import JobCard from '../../common/blocks/JobCard';
import Select from '../../common/components/Select';
import Hero from './Hero';

const HomeSection: React.FC = () => {
    const BUTTONS = [
        {
            title: 'Company',
            key: 'company' as const,
            options: [],
        },
        {
            title: 'Title',
            key: 'title' as const,
            options: [],
        },
        {
            title: 'Skills',
            key: 'company' as const,
            options: [],
        },
        {
            title: 'Job Type',
            key: 'jobType' as const,
            options: [],
        },
        {
            title: 'Location',
            key: 'location' as const,
            options: [
                {
                    name: 'Yes',
                    value: 'Yes',
                },
                {
                    name: 'No',
                    value: 'No',
                },
            ],
        },
    ];

    return (
        <div>
            <Hero />
            <section className="p-8">
                <h1 className="font-medium">Filter</h1>
                <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-5 gap-4 py-6">
                    {BUTTONS.map(({ title, key, options }) => (
                        <Select key={key} title={title} data={options} />
                    ))}
                </div>
                <div className="border border-gray-200 my-4" />
                <JobCard />
            </section>
        </div>
    );
};

export default HomeSection;
