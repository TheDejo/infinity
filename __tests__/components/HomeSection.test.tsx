import React from 'react';
import { render, Matcher, MatcherOptions, screen, act, fireEvent } from '@testing-library/react';
import HomeSection from '@/pageComponents/home/HomeSection';
import renderer from 'react-test-renderer';

const mockedPush = jest.fn();

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '/1',
            push: mockedPush,
        };
    },
}));

describe('Home Section  Component', () => {
    let getByTestId: (
        text: Matcher,
        options?: MatcherOptions | undefined,
        waitForElementOptions?: unknown,
    ) => HTMLElement;

    beforeEach(() => {
        const component = render(<HomeSection />);
        getByTestId = component.getByTestId;
    });

    it('should close the modal when the backdrop is clicked', () => {
        const applyBtn = screen.getAllByText('Apply Now')[0];
        expect(applyBtn).toBeInTheDocument();

        act(() => {
            fireEvent.click(applyBtn);
        });
        const modal = getByTestId('main-modal');
        expect(modal).toBeInTheDocument();

        const underlay = getByTestId('modal-underlay');
        expect(underlay).toBeInTheDocument();

        act(() => {
            fireEvent.mouseDown(underlay);
        });
        expect(modal).not.toBeInTheDocument();

        expect(underlay).not.toBeInTheDocument();
    });

    it('filters data when the search input is clicked', () => {
        const searchJob = getByTestId('hero-input') as HTMLInputElement;

        act(() => {
            fireEvent.change(searchJob, { target: { value: 'frontend developer' } });
        });
        expect(searchJob.value).toBe('frontend developer');

        const searchButton = screen.getByText('Find Jobs');

        act(() => {
            fireEvent.click(searchButton);
        });

        expect(screen.getAllByTestId('Frontend Developer')[0]).toBeInTheDocument();
    });

    it('pushes user to dynamic path when clicked', () => {
        const component = render(<HomeSection />);
        const button = screen.getAllByText('View Job')[0];
        act(() => {
            fireEvent.click(button);
        });

        expect(mockedPush).toBeCalledTimes(1);
        expect(mockedPush).toBeCalledWith('1');
    });

    it('matches Single input snapshot', () => {
        const tree = renderer.create(<HomeSection />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
