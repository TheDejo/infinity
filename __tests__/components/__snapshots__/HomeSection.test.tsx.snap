// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Section  Component matches Single input snapshot 1`] = `
<section
  data-testid="home-section"
>
  <section
    className="lg:px-8"
    test-dataid="hero-section"
  >
    <div
      className="p-8 bg-infinity-hero lg:rounded-[30px] flex justify-center items-center min-h-[30vh]"
    >
      <div
        className="mr-6"
      >
        <h1
          className="xl:w-[30rem] font-semibold text-2xl md:text-3xl lg:text-6xl mb-3"
        >
          Welcome To Infinity, & Beyond
        </h1>
        <p
          className="md:w-[30rem] font-normal text-sm md:text-base lg:text-xl"
        >
          Connect Today With Innovative Companies That Care About Planetary Diversity & Inclusion
        </p>
        <form
          className="flex flex-col md:flex-row mt-6 justify-start w-full"
        >
          <input
            className="rounded-lg  w-full md:w-72 xl:w-96 pl-6 py-2  placeholder-gray-600"
            data-testid="hero-input"
            name="search"
            onChange={[Function]}
            placeholder="Job Title or keywords"
            type="text"
            value=""
          />
          <button
            className="hidden md:flex justify-center items-center text-white text-lg rounded-lg bg-[#F3A494] px-5  shadow-infinity ml-2"
            onClick={[Function]}
          >
            Find Jobs
          </button>
        </form>
      </div>
      <figure
        className="hidden md:flex"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDc4IiBoZWlnaHQ9IjQzOCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <img
            alt="hero image"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="hero image"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fassets%2Fimages%2Fhero.png&w=1080&q=75"
              srcSet="/_next/image?url=%2Fassets%2Fimages%2Fhero.png&w=640&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fhero.png&w=1080&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </div>
      </figure>
    </div>
  </section>
  <div
    className="py-8"
  >
    <div
      className="px-8"
    >
      <h1
        className="font-medium"
      >
        Filter
      </h1>
      <div
        className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-5 gap-4 py-6"
      >
        <div
          className="bg-black text-sm py-3 px-4 flex justify-center items-center rounded-lg shadow-infinity outline-none border-none"
        >
          <select
            className="bg-black text-white w-full after:border-none"
            onChange={[Function]}
          >
            <option
              disabled={true}
              selected={true}
              value=""
            >
              Company
            </option>
            <option
              value="Vaarst"
            >
              Vaarst
            </option>
            <option
              value="Develop"
            >
              Develop
            </option>
            <option
              value="Dotmatics"
            >
              Dotmatics
            </option>
            <option
              value="Clowdr CIC"
            >
              Clowdr CIC
            </option>
            <option
              value="Econscia"
            >
              Econscia
            </option>
            <option
              value="Gear4Music"
            >
              Gear4Music
            </option>
            <option
              value="BBC"
            >
              BBC
            </option>
            <option
              value="Alliants"
            >
              Alliants
            </option>
            <option
              value="KOBAS"
            >
              KOBAS
            </option>
            <option
              value="Sainsburys"
            >
              Sainsburys
            </option>
          </select>
        </div>
        <div
          className="bg-black text-sm py-3 px-4 flex justify-center items-center rounded-lg shadow-infinity outline-none border-none"
        >
          <select
            className="bg-black text-white w-full after:border-none"
            onChange={[Function]}
          >
            <option
              disabled={true}
              selected={true}
              value=""
            >
              Title
            </option>
            <option
              value="Frontend Developer"
            >
              Frontend Developer
            </option>
            <option
              value="Backend Developer"
            >
              Backend Developer
            </option>
            <option
              value="Software Engineer"
            >
              Software Engineer
            </option>
            <option
              value="Junior Backend Developer"
            >
              Junior Backend Developer
            </option>
            <option
              value="Production Manager"
            >
              Production Manager
            </option>
            <option
              value="Software Developer"
            >
              Software Developer
            </option>
          </select>
        </div>
        <div
          className="bg-black text-sm py-3 px-4 flex justify-center items-center rounded-lg shadow-infinity outline-none border-none"
        >
          <select
            className="bg-black text-white w-full after:border-none"
            onChange={[Function]}
          >
            <option
              disabled={true}
              selected={true}
              value=""
            >
              Skills
            </option>
            <option
              value="TypeScript"
            >
              TypeScript
            </option>
            <option
              value="React"
            >
              React
            </option>
            <option
              value="Three.js"
            >
              Three.js
            </option>
            <option
              value="Typescript"
            >
              Typescript
            </option>
            <option
              value="Gatsby"
            >
              Gatsby
            </option>
            <option
              value="JavaScript"
            >
              JavaScript
            </option>
            <option
              value="GraphQL"
            >
              GraphQL
            </option>
            <option
              value="Django"
            >
              Django
            </option>
            <option
              value="Javascript"
            >
              Javascript
            </option>
            <option
              value="Management"
            >
              Management
            </option>
            <option
              value="Unit tests"
            >
              Unit tests
            </option>
            <option
              value="Git"
            >
              Git
            </option>
            <option
              value="Node Js"
            >
              Node Js
            </option>
            <option
              value="AWS"
            >
              AWS
            </option>
            <option
              value="PHP"
            >
              PHP
            </option>
            <option
              value="NodeJS"
            >
              NodeJS
            </option>
          </select>
        </div>
        <div
          className="bg-black text-sm py-3 px-4 flex justify-center items-center rounded-lg shadow-infinity outline-none border-none"
        >
          <select
            className="bg-black text-white w-full after:border-none"
            onChange={[Function]}
          >
            <option
              disabled={true}
              selected={true}
              value=""
            >
              Job Type
            </option>
            <option
              value="Full Time"
            >
              Full Time
            </option>
            <option
              value="Remote"
            >
              Remote
            </option>
            <option
              value="Parmanent"
            >
              Parmanent
            </option>
            <option
              value="Contract"
            >
              Contract
            </option>
          </select>
        </div>
        <div
          className="bg-black text-sm py-3 px-4 flex justify-center items-center rounded-lg shadow-infinity outline-none border-none"
        >
          <select
            className="bg-black text-white w-full after:border-none"
            onChange={[Function]}
          >
            <option
              disabled={true}
              selected={true}
              value=""
            >
              Location
            </option>
            <option
              value="Bristol BS1"
            >
              Bristol BS1
            </option>
            <option
              value="Bristol"
            >
              Bristol
            </option>
            <option
              value="Weschester"
            >
              Weschester
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
            <option
              value="London"
            >
              London
            </option>
            <option
              value="York YO26"
            >
              York YO26
            </option>
            <option
              value="London W1A"
            >
              London W1A
            </option>
            <option
              value="Southampton"
            >
              Southampton
            </option>
            <option
              value="Home Based"
            >
              Home Based
            </option>
            <option
              value="Edinburgh"
            >
              Edinburgh
            </option>
          </select>
        </div>
      </div>
      
    </div>
    <div
      className="border border-gray-200 my-4"
    />
    <section
      className="mx-8"
    >
      <div
        className="p-8 w-full min-h-[330px] border border-black rounded-3xl flex flex-col mb-8"
      >
        <div
          className="flex text-left w-full"
        >
          <h2
            className="font-medium text-2xl"
            data-testid="Frontend Developer"
          >
            Frontend Developer
          </h2>
        </div>
        <div
          className="border border-gray-200 my-4"
        />
        <div
          className="grid grid-cols-1 md:grid-cols-2 gap-5"
        >
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Company
              :
            </p>
            <p>
              Vaarst
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Description
              :
            </p>
            <p>
              We are looking for a skilled Frontend Developer to be a part our growing web platform team. You will be facing interesting problems that require skill
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Job Type
              :
            </p>
            <p>
              Full Time
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Skills
              :
            </p>
            <p>
              TypeScript, React, Three.js
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Location
              :
            </p>
            <p>
              Bristol BS1
            </p>
          </div>
        </div>
        <div
          className="border border-gray-200 my-4"
        />
        <div
          className="w-full flex md:flex-row flex-col justify-end"
        >
          <button
            className="px-6 py-2 mb-3 md:mb-0 mr-2 rounded-3xl hover:scale-110	flex justify-center items-center text-white bg-black"
            onClick={[Function]}
          >
            Apply Now
          </button>
          <button
            className="px-6 py-2 mb-3 md:mb-0 mr-2 rounded-3xl hover:scale-110	flex justify-center items-center text-black bg-[#FCF0EC]"
            onClick={[Function]}
          >
            View Job
          </button>
        </div>
      </div>
    </section>
    <section
      className="mx-8"
    >
      <div
        className="p-8 w-full min-h-[330px] border border-black rounded-3xl flex flex-col mb-8"
      >
        <div
          className="flex text-left w-full"
        >
          <h2
            className="font-medium text-2xl"
            data-testid="Frontend Developer"
          >
            Frontend Developer
          </h2>
        </div>
        <div
          className="border border-gray-200 my-4"
        />
        <div
          className="grid grid-cols-1 md:grid-cols-2 gap-5"
        >
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Company
              :
            </p>
            <p>
              Develop
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Description
              :
            </p>
            <p>
              We are recruiting on behalf of our client who are a global digital agency, who are looking to hire multiple React Developers on a contract basis for v
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Job Type
              :
            </p>
            <p>
              Remote, Parmanent
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Skills
              :
            </p>
            <p>
              React, Typescript, Gatsby
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Location
              :
            </p>
            <p>
              Bristol
            </p>
          </div>
        </div>
        <div
          className="border border-gray-200 my-4"
        />
        <div
          className="w-full flex md:flex-row flex-col justify-end"
        >
          <button
            className="px-6 py-2 mb-3 md:mb-0 mr-2 rounded-3xl hover:scale-110	flex justify-center items-center text-white bg-black"
            onClick={[Function]}
          >
            Apply Now
          </button>
          <button
            className="px-6 py-2 mb-3 md:mb-0 mr-2 rounded-3xl hover:scale-110	flex justify-center items-center text-black bg-[#FCF0EC]"
            onClick={[Function]}
          >
            View Job
          </button>
        </div>
      </div>
    </section>
    <section
      className="mx-8"
    >
      <div
        className="p-8 w-full min-h-[330px] border border-black rounded-3xl flex flex-col mb-8"
      >
        <div
          className="flex text-left w-full"
        >
          <h2
            className="font-medium text-2xl"
            data-testid="Frontend Developer"
          >
            Frontend Developer
          </h2>
        </div>
        <div
          className="border border-gray-200 my-4"
        />
        <div
          className="grid grid-cols-1 md:grid-cols-2 gap-5"
        >
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Company
              :
            </p>
            <p>
              Dotmatics
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Description
              :
            </p>
            <p>
              Collaborate with product management, application science and support stakeholders to develop and maintain product suite. Deliver high quality clean co
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Job Type
              :
            </p>
            <p>
              Remote
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Skills
              :
            </p>
            <p>
              JavaScript, React
            </p>
          </div>
          <div
            className="flex md:flex-row flex-col"
          >
            <p
              className="md:mr-3 mb-3 md:mb-0 w-24 text-darkBlue font-semibold"
            >
              Location
              :
            </p>
            <p>
              Weschester
            </p>
          </div>
        </div>
        <div
          className="border border-gray-200 my-4"
        />
        <div
          className="w-full flex md:flex-row flex-col justify-end"
        >
          <button
            className="px-6 py-2 mb-3 md:mb-0 mr-2 rounded-3xl hover:scale-110	flex justify-center items-center text-white bg-black"
            onClick={[Function]}
          >
            Apply Now
          </button>
          <button
            className="px-6 py-2 mb-3 md:mb-0 mr-2 rounded-3xl hover:scale-110	flex justify-center items-center text-black bg-[#FCF0EC]"
            onClick={[Function]}
          >
            View Job
          </button>
        </div>
      </div>
    </section>
    <div
      className="w-full flex justify-center items-center"
    >
      <button
        className="w-[40px] h-[40px] border border-[#c4c4c4] font-semibold rounded-md mr-2 hover:scale-110 text-white bg-black"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="w-[40px] h-[40px] border border-[#c4c4c4] font-semibold rounded-md mr-2 hover:scale-110"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="w-[40px] h-[40px] border border-[#c4c4c4] font-semibold rounded-md mr-2 hover:scale-110"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="w-[40px] h-[40px] border border-[#c4c4c4] font-semibold rounded-md mr-2 hover:scale-110"
        onClick={[Function]}
      >
        4
      </button>
    </div>
  </div>
</section>
`;
